sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSource.scala

```diff
@@ -559,6 +559,25 @@ case class DataSource(
       }
       globPath
     }.toSeq
+
+    if (checkFilesExist) {
+      val (filtered, filteredOut) = allGlobPath.partition { path =>
+        !InMemoryFileIndex.shouldFilterOut(path.getName)
+      }
+      if (filteredOut.nonEmpty) {
+        if (filtered.isEmpty) {
+          throw new AnalysisException(
+            "All path were ignored. The following path were ignored:\n" +
```

Comment:
path -> paths. Also, it seems clearer to say: "All paths were ignored:\n" and below, "Some paths were ignored:\n"

Reply:
I fixed it.